lib/rust/zephyr/src/lib.rs

```diff
@@ -24,8 +27,19 @@ use core::panic::PanicInfo;
 
 /// Override rust's panic.  This simplistic initial version just hangs in a loop.
 #[panic_handler]
-fn panic(_ :&PanicInfo) -> ! {
-    loop {
+fn panic(info :&PanicInfo) -> ! {
+    #[cfg(CONFIG_PRINTK)]
+    {
+        printkln!("panic: {}", info);
+    }
+    let _ = info;
```

Comment:
Alternatively, you can call the argument `_info` and use it in the `printkln!`.
```suggestion
fn panic(_info :&PanicInfo) -> ! {
    #[cfg(CONFIG_PRINTK)]
    {
        printkln!("panic: {}", _info);
    }
```

Reply:
Hmm, is there a typical way this is done? Both seem awkward.